packages/graphql-auth-transformer/src/resources.ts

```diff
@@ -217,7 +253,15 @@ groupsField: "${rule.groupsField || DEFAULT_GROUPS_FIELD}" }`
         formatComment?: (rule: AuthRule) => string,
     ) {
         let groupAuthorizationExpressions = []
+        let customClaim: string;
         for (const rule of rules) {
+            if (rule.groupClaim) {
+                if (customClaim) {
+                    throw new InvalidDirectiveError('@auth directive currently only supports one source for groupClaim!')
```

Comment:
This restriction isn't necessary if you set the userGroups as the first item (or after the comments) of the groupAuthorizationExpressions.
```typescript
groupAuthorizationExpressions = groupAuthorizationExpressions.concat(
  formatComment ?
    comment(formatComment(rule)) :
    comment(`Authorization rule: { allow: ${rule.allow}, groupsField: "${groupsAttribute}" }`),
  this.setUserGroups(rule.groupClaim),
  set(
     ref(variableToSet),
    raw(`$util.defaultIfNull($${variableToSet}, false)`)
  ),
```

assumes you adopt the setUserGroups overload, otherwise you can use a ternary to choose the right function to call.

Reply:
Ah I see each rule would have it's own respective claim. Though I'm not sure of the use case since the groups could exist in the same groupClaim. This might warrant a separate pr where we can discuss a use case for this.

Reply:
I already use this on my Amplify fork, and do have a use case. I have an auth rule based on the TenantId claim, and another auth rule based on a TeamId, and another AuthId based on the AppGroupId.
Only requires moving the setUserGroups to inside the forEach loop, so each rules evaluates against its own claimGroup. And then you can delete ~15 lines of code.

Reply:
@RossWilliams it's less about the code and more related to complexity. We're already adding new functionality here which is pretty flexible, and adding multiple claims that could overlap permissions is complexity on already a complex concept which could have downstream impact to the authorization rules which needs evaluation. At this time leaving this as-is seems to be a more prudent choice which we could re-evaluate as a separate feature in the future.